File: Examples/VisualBasic/NoteBook/WritingPasswordProtectedDoc.vb
Imports System.IO
Imports Aspose.Note
Imports Aspose.Note.Saving
Imports System.Drawing
Imports System.Globalization

Namespace WorkingWithNoteBook
    Public Class WritingPasswordProtectedDoc
        Public Shared Sub Run()
            ' ExStart:WritingPasswordProtectedDoc
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_NoteBook()

            Dim notebook = New Notebook(dataDir & Convert.ToString("notebook2.onetoc2"), New NotebookLoadOptions() With {
                 .DeferredLoading = False
            })

            notebook.Save(dataDir & Convert.ToString("notebook_out.onetoc2"), New NotebookOneSaveOptions() With {
                 .DeferredSaving = True
            })

            If Notebook.Count > 0 Then
                Dim childDocument0 = TryCast(Notebook(0), Document)

                childDocument0.Save(dataDir & Convert.ToString("Not Locked_out.one"))

                Dim childDocument1 = TryCast(Notebook(1), Document)

                childDocument1.Save(dataDir & Convert.ToString("Locked Pass1_out.one"), New OneSaveOptions() With {
                     .DocumentPassword = "pass"
                })

                Dim childDocument2 = TryCast(Notebook(2), Document)

                childDocument2.Save(dataDir & Convert.ToString("Locked Pass2_out.one"), New OneSaveOptions() With {
                     .DocumentPassword = "pass2"
                })
            End If
            ' ExEnd:WritingPasswordProtectedDoc       
        End Sub
    End Class
End Namespace

File: Examples/CSharp/NoteBook/WritingPasswordProtectedDoc.cs
using System.IO;
using Aspose.Note;
using Aspose.Note.Saving;
using System;
using System.Drawing;
using System.Globalization;

namespace Aspose.Note.Examples.CSharp.WorkingWithNoteBook
{
    using System.Linq;

    public class WritingPasswordProtectedDoc
    {
        public static void Run()
        {
            // ExStart:WritingPasswordProtectedDoc
            // ExFor:Notebook.RemoveChild
            // ExSummary:Shows how to save a notebook.

            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_NoteBook();

            var notebook = new Notebook(dataDir + "test.onetoc2", new NotebookLoadOptions() { DeferredLoading = false });

            notebook.Save(dataDir + "notebook_out.onetoc2", new NotebookOneSaveOptions() { DeferredSaving = true});

            if (notebook.Any())
            {
                var childDocument0 = notebook[0] as Document;

                childDocument0.Save(dataDir + "Not Locked_out.one");

                var childDocument1 = notebook[1] as Document;

                childDocument1.Save(dataDir + "Locked Pass1_out.one", new OneSaveOptions() { DocumentPassword = "pass" });

                var childDocument2 = notebook[2] as Document;

                childDocument2.Save(dataDir + "Locked Pass2_out.one", new OneSaveOptions() { DocumentPassword = "pass2" });
            }

            // ExEnd:WritingPasswordProtectedDoc       
        }
    }
}
